vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt

```diff
@@ -211,12 +212,23 @@ class WidgetPostAPIHandler @AssistedInject constructor(@Assisted private val roo
      * @param eventData the modular data
      */
     private fun getWidgets(widgetPostAPIMediator: WidgetPostAPIMediator, eventData: JsonDict) {
-        if (checkRoomId(widgetPostAPIMediator, eventData)) {
-            return
+        //if (checkRoomId(widgetPostAPIMediator, eventData)) {
+        //    return
+        //}
+        val roomIdInEvent = eventData["room_id"] as String?
```

Comment:
I'm wondering if this could be simplified to 

```kotlin
val allWidgets = when {
    eventData.isMissingRoomId() -> session.widgetService().getUserWidgets()
    roomId == eventData.roomId  -> session.widgetService().getRoomWidgets(roomId) + session.widgetService().getUserWidgets()
    else                        -> {
        widgetPostAPIMediator.sendError(stringProvider.getString(R.string.widget_integration_room_not_visible), eventData)
        return
    }
}
```

Reply:
That's looks nice, but i try your code in my ide, it looks like this
![image](https://user-images.githubusercontent.com/62228888/150759168-3f5c693b-b917-4ff2-822c-871ef496baed.png)

I'm sorry i'm really not familiar with kotlin, i just try to use the easy way i could understand~~

Reply:
ah my bad! this was a mix of pseudo code, these could be implemented as kotlin extension functions on the `JsonDict`, we could also extract another for matching against the room id  

```kotlin
private fun getWidgets(widgetPostAPIMediator: WidgetPostAPIMediator, eventData: JsonDict) {
  val allWidgets = when {
      eventData.isMissingRoomId()        -> session.widgetService().getUserWidgets()
      eventData.hasMatchingRoom(roomId)  -> session.widgetService().getRoomWidgets(roomId) + session.widgetService().getUserWidgets()
      else                               -> {
          widgetPostAPIMediator.sendError(stringProvider.getString(R.string.widget_integration_room_not_visible), eventData)
          return
      }
  }
  ... 
}

private fun JsonDict.isMissingRoomId() = (this["room_id"] as? String).isNullOrEmpty()
private fun JsonDict.hasMatchingRoom(roomId: String) = (this["room_id"] as? String) == roomId
```